### PR TITLE
Correcting EndpointUtils to display dependent endpoints

### PR DESCRIPTION
**Pull request by thushara35**
https://wso2.org/jira/browse/REGISTRY-2460

### CODE REVIEW

File: components/registry/org.wso2.carbon.registry.extensions/src/test/java/org/wso2/carbon/registry/extensions/test/EndpointUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+        String generatedEndpointcontent2 = EndpointUtils
+                .getEndpointContentWithOverview("http://www.webservicex.net/convertMetricWeight.asmx",
+                        "/_system/governance/trunk/endpoints/net/webservicex/www/ep-convertMetricWeight-asmx",
+                        EndpointUtils.deriveEndpointNameFromUrl("http://www.webservicex.net/convertMetricWeight.asmx"),
+                        "1.0.0");
+
+        assertTrue(generatedEndpointcontent1.equals(endpointContent1.toString()) || generatedEndpointcontent1
+                .equals(endpointContent1_2.toString()));
+        System.out.println("getEndpointContentWithOverview 1: " + EndpointUtils
```
Comment (shazni) [inline components/registry/org.wso2.carbon.registry.extensions/src/test/java/org/wso2/carbon/registry/extensions/test/EndpointUtilsTest.java]: Why do we need System.out.println in test cases? I think it's better to remove. Is this added for a reason?

File: components/registry/org.wso2.carbon.registry.extensions/src/test/java/org/wso2/carbon/registry/extensions/test/EndpointUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                .getEndpointContentWithOverview("http://www.webservicex.net/convertVolume.asmx",
+                        "/_system/governance/trunk/endpoints/net/webservicex/www/ep-convertVolume-asmx",
+                        EndpointUtils.deriveEndpointNameFromUrl("http://www.webservicex.net/convertVolume.asmx"),
+                        "1.0.0"));
+
+        assertTrue(generatedEndpointcontent2.equals(endpointContent2.toString()) || generatedEndpointcontent1
+                .equals(endpointContent2_2.toString()));
+        System.out.println("getEndpointContentWithOverview 1: " + EndpointUtils
```
Comment (shazni) [inline components/registry/org.wso2.carbon.registry.extensions/src/test/java/org/wso2/carbon/registry/extensions/test/EndpointUtilsTest.java]: Why do we need System.out.println in test cases? I think it's better to remove. Is this added for a reason?

File: components/registry/org.wso2.carbon.registry.extensions/src/main/java/org/wso2/carbon/registry/extensions/handlers/utils/EndpointUtils.java [hunk 3]
```diff
@@ -114,6 +121,12 @@ public static void saveEndpointsFromWSDL(String wsdlPath, Resource wsdlResource,
             log.error(msg, e);
```
Comment (shazni) [inline components/registry/org.wso2.carbon.registry.extensions/src/main/java/org/wso2/carbon/registry/extensions/handlers/utils/EndpointUtils.java]: Logging and throwing both done is unnecessary. I think just throwing the error with the message is sufficient.

File: components/registry/org.wso2.carbon.registry.extensions/src/test/java/org/wso2/carbon/registry/extensions/test/EndpointUtilsTest.java (hunk 2 above)
Comment (thushara35) [inline components/registry/org.wso2.carbon.registry.extensions/src/test/java/org/wso2/carbon/registry/extensions/test/EndpointUtilsTest.java]: For unit test we normally use SOP. I have observe this in other unit tests as well. Are you suggesting to use log.info on this?

File: components/registry/org.wso2.carbon.registry.extensions/src/test/java/org/wso2/carbon/registry/extensions/test/EndpointUtilsTest.java (hunk 1 above)
Comment (thushara35) [inline components/registry/org.wso2.carbon.registry.extensions/src/test/java/org/wso2/carbon/registry/extensions/test/EndpointUtilsTest.java]: For unit test we normally use SOP. I have observe this in other unit tests as well. Are you suggesting to use log.info on this?

File: components/registry/org.wso2.carbon.registry.extensions/src/main/java/org/wso2/carbon/registry/extensions/handlers/utils/EndpointUtils.java (hunk 3 above)
Comment (thushara35) [inline components/registry/org.wso2.carbon.registry.extensions/src/main/java/org/wso2/carbon/registry/extensions/handlers/utils/EndpointUtils.java]: Hi, 
Thanks for bringing this up will remove the log.

File: components/registry/org.wso2.carbon.registry.extensions/src/test/java/org/wso2/carbon/registry/extensions/test/EndpointUtilsTest.java (hunk 1 above)
Comment (shazni) [inline components/registry/org.wso2.carbon.registry.extensions/src/test/java/org/wso2/carbon/registry/extensions/test/EndpointUtilsTest.java]: You can't have SOP anywhere. It's bad practice. Why do you need SOP or even a logging in the first place in a test case. Test cases are for validations.

Comment (thushara35) [inline components/registry/org.wso2.carbon.registry.extensions/src/test/java/org/wso2/carbon/registry/extensions/test/EndpointUtilsTest.java]: Understood. Will remove the SOPs

File: components/registry/org.wso2.carbon.registry.extensions/src/test/java/org/wso2/carbon/registry/extensions/test/EndpointUtilsTest.java (hunk 2 above)
Comment (thushara35) [inline components/registry/org.wso2.carbon.registry.extensions/src/test/java/org/wso2/carbon/registry/extensions/test/EndpointUtilsTest.java]: Understood. Will remove the SOPs

**Pull request merged**